src/main/java/core/basesyntax/service/impl/OperationProcessorImpl.java

```diff
@@ -0,0 +1,45 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.dao.FruitDao;
+import core.basesyntax.dao.impl.FruitDaoImpl;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.OperationProcessor;
+import core.basesyntax.strategy.OperationHandler;
+import core.basesyntax.strategy.OperationStrategy;
+import core.basesyntax.strategy.impl.BalanceOperationHandler;
+import core.basesyntax.strategy.impl.PurchaseOperationHandler;
+import core.basesyntax.strategy.impl.ReturnOperationHandler;
+import core.basesyntax.strategy.impl.SupplyOperationHandler;
+import java.util.Map;
+
+public class OperationProcessorImpl implements OperationProcessor {
+    private static final Map<FruitTransaction.Operation, OperationHandler> operationHandlerMap;
+    private static FruitDao fruitDao = new FruitDaoImpl();
+    private static OperationStrategy operationStrategy;
+
+    static {
+        operationHandlerMap = 
+            Map.of(FruitTransaction.Operation.BALANCE, new BalanceOperationHandler(fruitDao),
+                FruitTransaction.Operation.PURCHASE, new PurchaseOperationHandler(fruitDao),
+                FruitTransaction.Operation.SUPPLY, new SupplyOperationHandler(fruitDao),
+                FruitTransaction.Operation.RETURN, new ReturnOperationHandler(fruitDao));
+    }
+
+    public OperationProcessorImpl() {
+    }
+
+    public OperationProcessorImpl(FruitDao fruitDao, OperationStrategy operationStrategy) {
+        this.fruitDao = fruitDao;
+        this.operationStrategy = operationStrategy;
+    }
```

Comment:
```suggestion
    private FruitDao fruitDao;
    private OperationStrategy operationStrategy;

    public OperationProcessorImpl(FruitDao fruitDao, OperationStrategy operationStrategy) {
        this.fruitDao = fruitDao;
        this.operationStrategy = operationStrategy;
    }
```

Reply:
why dao and strategy are static?
map should be created and passed to OperAtionStrategy in main, you don't need it here

Reply:
The map was originally in main but i was told to move it to another class, now i have to put it back in main?